### PR TITLE
AR_Motors: fix have_skid_steering to return true for omnis

### LINKED ISSUE #21600: Rover: has_skid_steering should report true for omni vehicles
Omni rovers and boats can pivot just like more normal skid-steering vehicles.

We should enhancement the [AR_Motors::have_skid_steering()](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AR_Motors/AP_MotorsUGV.cpp#L267) method so that it returns true for omni vehicles.

This was first reported here https://github.com/ArduPilot/ardupilot_wiki/issues/4553

### PR DESCRIPTION
**Pull request by shiv-tyagi**
Fixes #21600. @rmackay9 Please check if this is the change you were looking for.

### CODE REVIEW

File: libraries/AR_Motors/AP_MotorsUGV.cpp [hunk 1]
```diff
@@ -266,11 +266,13 @@ float AP_MotorsUGV::get_slew_limited_throttle(float throttle, float dt) const
...
-    if (SRV_Channels::function_assigned(SRV_Channel::k_throttleLeft) &&
-        SRV_Channels::function_assigned(SRV_Channel::k_throttleRight)) {
-        return true;
-    }
-    return false;
+    bool simple_skid_steering = SRV_Channels::function_assigned(SRV_Channel::k_throttleLeft) && SRV_Channels::function_assigned(SRV_Channel::k_throttleRight);
+    bool omni = SRV_Channels::function_assigned(SRV_Channel::k_motor1) && 
+                SRV_Channels::function_assigned(SRV_Channel::k_motor2) &&
+                SRV_Channels::function_assigned(SRV_Channel::k_motor3) && 
+                SRV_Channels::function_assigned(SRV_Channel::k_motor4);
```
Comment (IamPete1) [inline libraries/AR_Motors/AP_MotorsUGV.cpp:273]: We do support 3 motor omni, I think you could just check `_motors_num > 0`.

I did a little rework in this area in https://github.com/ArduPilot/ardupilot/pull/18612

Comment (shiv-tyagi) [reply libraries/AR_Motors/AP_MotorsUGV.cpp]: Thanks @IamPete1 for letting me know. Please check if I have done the right thing now.

**Comment by rmackay9** [comment]
@shiv-tyagi,

Thanks for this.  I'd prefer if we had an is_omni() method a bit like @IamPete1's change had and then use it wherever we need to make the determination of whether this is an omni vehicle or not.   For now I think this is, "init", "has_skid_steering", "output_omni".  I just worry that otherwise we could end up with a different result in the two or three places leading to odd behaviour.

I'm not sure that checking for num_motors > 0 is sufficient.

A couple of side notes:

- I think we should remove this check completely from init(), "if (_frame_type != FRAME_TYPE_UNDEFINED) {"
- setup_safety_output() may not be configuring the safety outputs for omni motors. <-- scope creep, don't need to do for this PR

**Comment by shiv-tyagi** [comment]
@rmackay9 
Thanks for your time to review this PR. I have added the `is_omni()` method as requested by you. Please have a look at the PR and let me know if it is mergeable now. If anything else needs to be changed, please let me know. I'll be happy to do it.
Thanks again!

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**